File: service/src/main/java/by/rekuts/travelagency/service/CountryService.java
package by.rekuts.travelagency.service;

import by.rekuts.travelagency.domain.Country;
import by.rekuts.travelagency.service.impl.CrudService;

/**
 * Interface CountryService concerns to Service layer.
 * Realisations of this interface has access to DAO layer
 * in the face of CountryRepository interface and could call its methods.
 */
public interface CountryService extends CrudService<Country> {
}

File: service/src/main/scala/by/rekuts/travelagency/service/CountryService.scala
package by.rekuts.travelagency.service

import by.rekuts.travelagency.domain.Country

/**
  * Trait CountryService concerns to Service layer.
  * Realisations of this trait has access to DAO layer
  * in the face of CountryRepository trait and could call its methods.
  */
trait CountryService extends CrudService[Country]
